package rpc_eterm;


message EmptyParams {

}

message Status {

  optional bool connected = 1;
  optional bool busy = 2;
}

message AccessTime {

  optional int32 seconds = 1;
  optional int32 nanoseconds = 2;
}

message UserName {

  optional string first_name = 1;
  optional string last_name  = 2;
}

message User {

  optional int32 user_id      = 1;
  optional UserName user_name  = 2;
  optional uint32 access_count = 4;
  optional AccessTime last_access = 5;
}

message UserList {

  repeated User   users  = 2;
}

message UserID {

  optional uint32 user_id = 1;
}

message HcSetLedState {

  optional uint32 led_id = 1;
  optional bool   led_on = 2;
}

